packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -203,10 +209,11 @@ export const LocationDetailViewTable = (): JSX.Element | null => {
 
   const { direction, selection } = sortState;
 
-  const tableData =
+  const tableData = (
     direction === 'ascending'
       ? data.result.sort((a, b) => compareFn(a[selection], b[selection]))
-      : data.result.sort((a, b) => compareFn(b[selection], a[selection]));
+      : data.result.sort((a, b) => compareFn(b[selection], a[selection]))
+  ).slice(start, end);
```

Comment:
Need to run `slice` before `sort`, else the rendered list will be non-deterministic

Reply:
Thanks, updated in latest.